File: examples/Variables.java
// --------------
// Variables.java
// --------------

import java.util.Arrays;

final class Variables {
    public static void main (String[] args) {
        System.out.println("Variables.java");

        {
        int i = 2;
        int v = i;
        ++v;
        assert i == 2;
        assert v == 3;
        }

        {
        Integer i = 2;
        Integer v = i;
        ++v;
        assert i == 2;
        assert v == 3;
        }

        {
        final int[] a = {2, 3, 4};
        assert a.length == 3;
        assert a[1]     == 3;
        try {
            assert a[3] == 0;
            assert false;}
        catch (ArrayIndexOutOfBoundsException e)
            {}
        }

        {
        final int[] a = {2, 3, 4};
        final int[] b = a;
        assert a == b;
        }

        {
        final int[] a = {2, 3, 4};
        final int[] b = {2, 3, 4};
        assert a != b;
        assert !a.equals(b);
        assert Arrays.equals(a, b);
        }

        {
        final int[] a = {2, 3, 4};
        final int[] b = a;
        ++b[1];
        assert a[1] == 4;
        assert b[1] == 4;
        }

        System.out.println("Done.");}}

File: examples/Variables.js
// -------------
// Variables.js
// -------------

function assert (b) {
    if (!b) {
        throw "Assertion Error";}}

print("Variables.js\n");

i = 2;
v = i;
assert(i == v);
v += 1;
assert(i == 2);
assert(v == 3);

a = [2, 3, 4];
b = a;
assert(a == b);
a[1] += 1;
assert(a[1] == 4);
assert(b[1] == 4);

a = [2, 3, 4];
b = [2, 3, 4];
assert(a != b);
assert(!(a < b) || (a > b));

a = [2, 3, 4];
b = a;
assert(a == b);
a.push(5);
assert(a == b);
assert(a != [2, 3, 4, 5])
assert(!((a < [2, 3, 4, 5]) || (a > [2, 3, 4, 5])));

print("Done.\n");
